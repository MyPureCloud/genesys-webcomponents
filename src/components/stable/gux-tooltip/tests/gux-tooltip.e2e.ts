import { newE2EPage, E2EPage } from '@stencil/core/testing';

function sleep(ms: number) {
  return new Promise(resolve => setTimeout(resolve, ms));
}

async function newDeterministicE2EPage({
  html
}: {
  html: string;
}): Promise<E2EPage> {
  const page = await newE2EPage();

  await page.evaluateOnNewDocument(() => {
    Math.random = () => 0.5;
  });
  await page.setContent(html);
  await page.waitForChanges();

  return page;
}

describe('gux-tooltip', () => {
  describe('#render', () => {
    [
      `
        <div lang="en">
          <div>Element</div>
          <gux-tooltip>Tooltip</gux-tooltip>
        </div>
      `,
      `
        <div lang="en">
          <div id="element">Element</div>
          <gux-tooltip for="element">Tooltip</gux-tooltip>
        </div>
      `
    ].forEach((html, index) => {
      it(`should render component as expected (${index + 1})`, async () => {
        const page = await newDeterministicE2EPage({ html });
        const element = await page.find('div[lang]');

        expect(element.innerHTML).toMatchSnapshot();
      });
    });
  });
});
