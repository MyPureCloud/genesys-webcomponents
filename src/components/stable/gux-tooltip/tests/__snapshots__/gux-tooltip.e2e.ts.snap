// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`gux-tooltip #render should render component as expected (1) 1`] = `" <div>Element</div> <gux-tooltip id=\\"gux-tooltip-i\\" class=\\"hydrated\\" tabindex=\\"0\\" role=\\"tooltip\\" data-popper-placement=\\"bottom-start\\" style=\\"position: fixed; left: 0px; top: 0px; margin: 0px; right: auto; bottom: auto; transform: translate(8px, 44px);\\"><!---->Tooltip</gux-tooltip> "`;

exports[`gux-tooltip #render should render component as expected (2) 1`] = `" <div id=\\"element\\" class=\\"gux-tooltip-for-element\\" aria-describedby=\\"gux-tooltip-i\\">Element</div> <gux-tooltip for=\\"element\\" id=\\"gux-tooltip-i\\" class=\\"hydrated\\" tabindex=\\"0\\" role=\\"tooltip\\" data-popper-placement=\\"bottom-start\\" style=\\"position: fixed; left: 0px; top: 0px; margin: 0px; right: auto; bottom: auto; transform: translate(8px, 44px);\\"><!---->Tooltip</gux-tooltip> "`;
